From: Lars Ellenberg <lars.ellenberg@linbit.com>
Date: Mon, 14 Sep 2009 11:33:26 +0000 (+0200)
Subject: compat: remove special -I$(KDIR)/include for drbdsetup
X-Git-Url: http://git.drbd.org/?p=drbd-8.3.git;a=commitdiff_plain;h=a6e36c389adb4485313191fd50405395fa3c1a74

compat: remove special -I$(KDIR)/include for drbdsetup

Finally get rid of that special case hack.
---

diff --git a/user/Makefile b/user/Makefile
index 04cdcae..0143e09 100644
--- a/user/Makefile
+++ b/user/Makefile
@@ -22,8 +22,6 @@ KDIR := /lib/modules/$(shell uname -r)/build
 CC = gcc
 OPTFLAGS = -O1
 CFLAGS = $(OPTFLAGS) -c -W -Wall -I../drbd
-# for the netlink connector stuff <linux/connector.h>:
-drbdsetup: CFLAGS += -I$(KDIR)/include
 
 drbdadm-obj = drbdadm_scanner.o drbdadm_parser.o drbdadm_main.o \
 	      drbdadm_adjust.o drbdtool_common.o drbdadm_usage_cnt.o \
diff --git a/user/drbdsetup.c b/user/drbdsetup.c
index 4c3d132..d3b353d 100644
--- a/user/drbdsetup.c
+++ b/user/drbdsetup.c
@@ -46,6 +46,7 @@
 #include <time.h>
 
 #define __bitwise /* Build-workaround for broken RHEL4 kernels (2.6.9_78.0.1) */
+#include <linux/types.h>
 #include <linux/netlink.h>
 #include <linux/connector.h>
 
